src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -644,6 +644,90 @@ protected override void ApplyAnalyzerReferenceRemoved(ProjectId projectId, Analy
             }
         }
 
+        protected override void ApplyProjectChanges(ProjectChanges projectChanges)
```

Comment:
What happens if two projects both want to edit the same Razor file. Can that happen?

Reply:
Yes that could happen (a razor file is included in multiple projects).  However the invisible editor requires me to pass the project, so I believe I'd still need to open the invisible editor twice, once for each project.